diff --git a/drivers/wifi/ameba/ameba_wifi.h b/drivers/wifi/ameba/ameba_wifi.h
--- a/drivers/wifi/ameba/ameba_wifi.h
+++ b/drivers/wifi/ameba/ameba_wifi.h
@@ -81,6 +81,7 @@ enum rtk_state_flag {
 };
 
 void wifi_init(void);
+int eth_rtk_rx(uint8_t idx, void *buffer, uint16_t len);
 int wifi_get_scan_records(unsigned int *AP_num, char *scan_buf);
 int wifi_get_mac_address(int idx, struct _rtw_mac_t *mac, uint8_t efuse);
 int rltk_wlan_send(int idx, void *pkt_addr, uint32_t len);
diff --git a/drivers/wifi/ameba/ameba_wifi_drv.c b/drivers/wifi/ameba/ameba_wifi_drv.c
--- a/drivers/wifi/ameba/ameba_wifi_drv.c
+++ b/drivers/wifi/ameba/ameba_wifi_drv.c
@@ -107,10 +107,15 @@ static int ameba_wifi_send(const struct device *dev, struct net_pkt *pkt)
 }
 
 /* should called in driver after rx */
-static int eth_rtk_rx(uint8_t idx, void *buffer, uint16_t len)
+int eth_rtk_rx(uint8_t idx, void *buffer, uint16_t len)
 {
 	struct net_pkt *pkt;
 
+	if (idx >= ARRAY_SIZE(ameba_wifi_iface)) {
+		LOG_ERR("invalid network interface index: %u", idx);
+		return -EINVAL;
+	}
+
 	if (ameba_wifi_iface[idx] == NULL) {
 		LOG_ERR("network interface unavailable");
 		return -EIO;
